src/core/lib/security/credentials/credentials.h

```diff
@@ -181,28 +182,22 @@ using CredentialsMetadataArray = std::vector<std::pair<Slice, Slice>>;
 
 /* --- grpc_call_credentials. --- */
 
-namespace grpc_core {
-
-// Abstract interface to obtain contextual data for some call credentials.
-class AuthMetadataContext {
- public:
-  virtual std::string JwtServiceUrl(
-      const ClientInitialMetadata& metadata) const = 0;
-  virtual grpc_auth_metadata_context MakeLegacyContext(
-      const ClientInitialMetadata& metadata) const = 0;
-
- protected:
-  ~AuthMetadataContext() = default;
-};
-
-}  // namespace grpc_core
-
 // This type is forward declared as a C struct and we cannot define it as a
 // class. Otherwise, compiler will complain about type mismatch due to
 // -Wmismatched-tags.
 struct grpc_call_credentials
     : public grpc_core::RefCounted<grpc_call_credentials> {
  public:
+  // TODO(roth): Consider whether security connector actually needs to
```

Comment:
What if we make this a member of the filter, so we're not rebuilding it every request but just passing out a const ref to the creds.

Reply:
The perils of phone based code review!